addons/account/models/account_move.py

```diff
@@ -1781,10 +1781,49 @@ def _get_tax_cash_basis_base_account(self, line, tax):
     def _get_amount_tax_cash_basis(self, amount, line):
         return line.company_id.currency_id.round(amount)
 
+    def _get_tax_cash_basis_hash(self, tax, move, line):
+        account_id = self._get_tax_cash_basis_base_account(line, tax)
+        return (move.id, account_id.id, tax.id, line.currency_id.id, line.partner_id.id)
+
+    def _get_tax_cash_basis_base_line(self, key, new_move):
+        move, account_id, tax, currency_id, partner_id = key
+        move = self.env['account.move'].browse(move)
+        return {
+            'name': move.name,
+            'account_id': account_id,
+            'tax_exigible': True,
+            'tax_ids': [(6, 0, [tax])],
+            'move_id': new_move.id,
+            'currency_id': currency_id,
+            'partner_id': partner_id,
+        }
+
+    def _create_tax_cash_basis_base_line(self, keys, amount_dict, amount_currency_dict, new_move):
+        for key in keys:
+            base_line = self._get_tax_cash_basis_base_line(key, new_move)
+            currency_id = base_line.get('currency_id', False)
+            rounded_amt = amount_dict[key]
+            amount_currency = amount_currency_dict[key] if currency_id else 0.0
+            aml_obj = self.env['account.move.line'].with_context(check_move_validity=False)
+            aml_obj.create(dict(
+                base_line,
+                debit=rounded_amt > 0 and rounded_amt or 0.0,
+                credit=rounded_amt < 0 and abs(rounded_amt) or 0.0,
+                amount_currency=amount_currency))
+            aml_obj.create(dict(
+                base_line,
+                credit=rounded_amt > 0 and rounded_amt or 0.0,
+                debit=rounded_amt < 0 and abs(rounded_amt) or 0.0,
+                amount_currency=-amount_currency,
+                tax_ids=[]))
+
     def create_tax_cash_basis_entry(self, percentage_before_rec):
         self.ensure_one()
         move_date = self.debit_move_id.date
         newly_created_move = self.env['account.move']
+        caba_amount_dict = {}
+        caba_amount_currency_dict = {}
+        caba_keys = set()
```

Comment:
This variable seems useless. These keys are anyway always in caba_amount_dict, aren't they?